src/main/scala/com/fasterxml/jackson/module/scala/modifiers/ScalaTypeModifier.scala

```diff
@@ -0,0 +1,33 @@
+package com.fasterxml.jackson.module.scala.modifiers
+
+import java.lang.reflect.{ParameterizedType, Type}
+
+import com.fasterxml.jackson.databind.JavaType
+import com.fasterxml.jackson.databind.`type`._
+import com.fasterxml.jackson.module.scala.JacksonModule
+
+class ScalaTypeModifier extends TypeModifier {
+  override def modifyType(javaType: JavaType,
+                          jdkType: Type,
+                          context: TypeBindings,
+                          typeFactory: TypeFactory): JavaType = {
+
+
+    if (javaType.isTypeOrSubTypeOf(classOf[collection.Map[_, _]])) {
+      MapLikeType.upgradeFrom(javaType, javaType.containedTypeOrUnknown(0), javaType.containedTypeOrUnknown(1))
```

Comment:
Exact type checking may or may not work. I believe an issue I had not-too-long ago was that the type hierarchy hit Collection before Map. I don't remember the specifics to make it work, but might be something you'll run into (and it's nice to have a heads up as you start digging around that that could be an issue).